### ISSUE TITLE
wfs:wfs-1.1.0-Basic-GetCapabilities-tc7 fails with InvocationTargetException

### ISSUE THREAD

**Issue by lgoltz** [opengeospatial/ets-wfs11#78]
Trying to test our WFS server here and get the following result:

URL: http://192.168.100.165:9090/geoserver/test/wfsversion=1.1.0#request=GetCapabilities,service=WFS
Error in call to extension function {public org.w3c.dom.NodeList com.occamlab.te.TECore.request(org.w3c.dom.Document,java.lang.String) throws java.lang.Throwable}: Exception in extension function java.lang.reflect.InvocationTargetException
Result: Failed

Orginal comment in https://github.com/opengeospatial/ets-wms11/issues/34#issuecomment-357861689

**Comment by lgoltz** [comment]
@sprd16 Can you provide the capabilities URL of the service under test? What's the name of the failing test?

**Comment by sprd16** [comment]
Test wfs:wfs-1.1.0-Basic-GetCapabilities-tc7

**Comment by lgoltz** [comment]
@sprd16 If the service is not public, please download the session and provide them to us.

**Comment by sprd16** [comment]
GetCapabilities session or error log

**Comment by lgoltz** [comment]
The test session you can download after testing. But of course an error log could be helpful, too. 
I assume you are testing locally?

**Comment by sprd16** [comment]
I downloaded Teamengine and setup it locally and got WFS-Test suite data from below url: 
http://cite.opengeospatial.org/teamengine/about/wfs/1.1.0/site/index.html

When running my geoserver with test data, facing the problem
Please find the attached log file
[log.txt](https://github.com/opengeospatial/ets-wfs11/files/1634141/log.txt)

**Comment by lgoltz** [comment]
There could be a problem with the missing question mark in the URL used for testing. Instead of http://192.168.100.165:9090/geoserver/test/wfsversion=1.1.0#request=GetCapabilities,service=WFS it should be http://192.168.100.165:9090/geoserver/test/wfs?version=1.1.0#request=GetCapabilities,service=WFS.

Can you change the configuration of the DCP URL for GetCapabilities GET in your capabilities and append the question mark?

**Comment by sprd16** [comment]
Method: get
   URL: http://192.168.100.165:9090/geoserver/sf/wfsversion=1.1.0#request=GetCapabilities,service=WFS
Error in call to extension function {public org.w3c.dom.NodeList com.occamlab.te.TECore.request(org.w3c.dom.Document,java.lang.String) throws java.lang.Throwable}: Exception in extension function java.lang.reflect.InvocationTargetException
Result: Failed

**Comment by sprd16** [comment]
I gave the URL: 
http://192.168.100.165:9090/geoserver/sf/wfs?service=WFS&version=1.1.0&request=GetCapabilities .

But at wfs-1.1.0-Basic-GetCapabilities-tc7

Method: get
URL: http://192.168.100.165:9090/geoserver/sf/wfsversion=1.1.0#request=GetCapabilities,service=WFS

Error in call to extension function {public org.w3c.dom.NodeList com.occamlab.te.TECore.request(org.w3c.dom.Document,java.lang.String) throws java.lang.Throwable}: Exception in extension function java.lang.reflect.InvocationTargetException
Result: Failed

**Comment by lgoltz** [comment]
Here the description of the test `wfs-1.1.0-Basic-GetCapabilities-tc7`:
```
[1] GetCapabilities with the proper KVP syntax. Pass if all of the following conditions are true: 
  (1) the response is schema valid; (2) the root document is an wfs:WFS_Capabilities document.
[2] GetCapabilities with improper KVP syntax (using '#' and ',' instead of '?' or '&amp;'). Pass if all of the
      following conditions are true: 
  (1) the response is schema valid; (2) the root document is an ows:ExceptionReport document
```
So the URL http://192.168.100.165:9090/geoserver/sf/wfsversion=1.1.0#request=GetCapabilities,service=WFS is expected with the exception of the missing question mark as described in https://github.com/opengeospatial/ets-wfs11/issues/78#issuecomment-357887729

**Comment by sprd16** [comment]
Yes. 
http://192.168.100.165:9090/geoserver/sf/wfs? (question mark) is missing here
version=1.1.0#request=GetCapabilities,service=WFS 

Do I need to do any changes in geoserver WFS settings?

**Comment by sprd16** [comment]
Please find my geoserver settings 
![geoserver_wfs](https://user-images.githubusercontent.com/35480183/34986532-ef61e1e2-fadd-11e7-8530-094c3fb98bc2.PNG)

**Comment by lgoltz** [comment]
As far as I understand from OWS Common 1.0 (http://portal.opengeospatial.org/files/?artifact_id=8798) the HTTP GET URL must end with an question mark '?' or an ampersand '&' (see 11.2 HTTP GET). But as WFS 1.1.0 references OWS Common 0.3 I cannot ensure this is true for WFS 1.1.0.  This means I'm not sure if it is a failure in the ETS or not.
I'm not sure if the missing question mark is the cause of the failure. So we have to made some more investigations here. 

@sprd16 Sorry, I cannot help you with your geoserver setting. But maybe you can post the response from the URL without question mark (http://192.168.100.165:9090/geoserver/sf/wfsversion=1.1.0#request=GetCapabilities,service=WFS).

**Comment by sprd16** [comment]
Sorry, delay in replying.

If I check the URL without question mark , i am getting 404 error.

Now i am trying with WFS1.0.0 test data and it is failing at wfs:test1.0.0-basic-getfeature-filter-comparisonoperators-get-1 through 25

**Comment by lgoltz** [comment]
> Now i am trying with WFS1.0.0 test data and it is failing at wfs:test1.0.0-basic-getfeature-filter-comparisonoperators-get-1 through 25

@sprd16 Please create a new issue in the issue tracker of the ETS of the service you are testing  (https://github.com/opengeospatial/ets-wfs10/issues) and provide some more information.

**Comment by lgoltz** [comment]
Fixed with commit https://github.com/opengeospatial/ets-wfs11/pull/79/commits/2790201e6ff640f716aed13a37613a09f26e1e2b, PR #79

**Comment by lgoltz** [comment]
Fix was replaced by PR #80
